lib/internal/modules/esm/loader.js

```diff
@@ -70,27 +100,35 @@ class ESMLoader {
   /**
    * Prior to ESM loading. These are called once before any modules are started.
    * @private
-   * @property {Function[]} globalPreloaders First-in-first-out list of
-   * preload hooks.
+   * @property {KeyedHook[]} globalPreloaders Last-in-first-out
+   *  list of preload hooks.
    */
   #globalPreloaders = [];
 
   /**
    * Phase 2 of 2 in ESM loading.
    * @private
-   * @property {Function[]} loaders First-in-first-out list of loader hooks.
+   * @property {KeyedHook[]} loaders Last-in-first-out
+   *  collection of loader hooks.
    */
   #loaders = [
-    defaultLoad,
+    ObjectAssign(ObjectCreate(null), {
+      fn: defaultLoad,
+      url: 'node:lib/internal/modules/esm/load.js',
+    }),
   ];
 
   /**
    * Phase 1 of 2 in ESM loading.
    * @private
-   * @property {Function[]} resolvers First-in-first-out list of resolver hooks
+   * @property {KeyedHook[]} resolvers Last-in-first-out
+   *  collection of resolver hooks.
    */
   #resolvers = [
-    defaultResolve,
+    ObjectAssign(ObjectCreate(null), {
+      fn: defaultResolve,
+      url: 'node:lib/internal/modules/esm/resolve.js',
+    }),
```

Comment:
Do we need a null-prototype here?
```suggestion
    {
      fn: defaultResolve,
      url: 'node:lib/internal/modules/esm/resolve.js',
    },
```

Reply:
I suppose no 👍